src/tiledb/cloud/file_ingestion/ingestion.py

```diff
@@ -0,0 +1,177 @@
+from math import ceil
+from typing import Any, List, Mapping, Optional, Sequence, Union
+
+import tiledb
+from tiledb.cloud import dag
+from tiledb.cloud.dag.mode import Mode
+from tiledb.cloud.utilities import as_batch
+from tiledb.cloud.utilities import get_logger_wrapper
+from tiledb.cloud.vcf.ingestion import find_uris_udf
+
+DEFAULT_RESOURCES = {"cpu": "2", "memory": "8Gi"}
+DEFAULT_FILE_INGESTION_NAME = "file-ingestion"
+DEFAULT_BATCH_SIZE = 100
+
+
+def chunk_results_udf(
+    udf_results: Sequence[Sequence[str]], batch_size: Optional[int] = DEFAULT_BATCH_SIZE
+) -> List[List[str]]:
+    """
+    Flatten and break a list of udf results into batches of a specified size.
+
+    :param udf_results: An iterable of iterables containing UDF results.
+    :param batch_size: Resulting chunk size, defaults to 100.
+    :return List[List[str]]: A list of chunks as lists.
+    """
+    # Reduce batch size if there are fewer sample URIs
+    batch_size = min(batch_size, len(udf_results))
+    num_chunks = ceil(len(udf_results) / batch_size)
+
+    flattened_results = [result for udf_result in udf_results for result in udf_result]
+    return [
+        flattened_results[n * batch_size : (n + 1) * batch_size]
```

Comment:
Do we have to check if we go beyond the size of `flattened_results`?

Reply:
Not really because even if we go out of size while slicing, the last slice will be as big as possible without errors.